2021-01-11  Jakub Jelinek  <jakub@redhat.com>
 
        * configure.ac: Ensure c/Make-lang.in comes first in @all_lang_makefrags@.
        * configure: Regenerated.
 
--- a/src/gcc/configure.ac.jj 2021-01-05 13:57:59.911905006 +0100
+++ b/src/gcc/configure.ac    2021-01-11 14:49:22.878218318 +0100
@@ -6977,7 +6977,12 @@
 	$ok || continue
 
 	all_lang_configurefrags="$all_lang_configurefrags \$(srcdir)/$gcc_subdir/config-lang.in"
-	all_lang_makefrags="$all_lang_makefrags \$(srcdir)/$gcc_subdir/Make-lang.in"
+	if test "x$language" = xc && test -n "$all_lang_makefrags"; then
+	    # Put c/Make-lang.in fragment first to match serialization languages order.
+	    all_lang_makefrags="\$(srcdir)/$gcc_subdir/Make-lang.in $all_lang_makefrags"
+	else
+	    all_lang_makefrags="$all_lang_makefrags \$(srcdir)/$gcc_subdir/Make-lang.in"
+	fi
 	if test -f $srcdir/$gcc_subdir/lang.opt; then
 	    lang_opt_files="$lang_opt_files $srcdir/$gcc_subdir/lang.opt"
 	    all_opt_files="$all_opt_files $srcdir/$gcc_subdir/lang.opt"
